mono/metadata/icall.c

```diff
@@ -6585,11 +6585,11 @@ char **environ;
 #endif
 #endif
 
-ICALL_EXPORT MonoArray *
-ves_icall_System_Environment_GetCommandLineArgs (void)
+ICALL_EXPORT MonoArrayHandle
+ves_icall_System_Environment_GetCommandLineArgs (MonoError *error)
 {
-	MonoError error;
-	MonoArray *result = mono_runtime_get_main_args_checked (&error);
+	error_init (error);
+	MonoArrayHandle result = mono_runtime_get_main_args_handle (error);
 	mono_error_set_pending_exception (&error);
```

Comment:
icalls that have been converted to use handles, don't need to call `mono_error_set_pending_exception`: the wrapper will do it on the icall's behalf.